src/api/java/com/minecolonies/api/colony/interactionhandling/AbstractInteractionResponseHandler.java

```diff
@@ -0,0 +1,155 @@
+package com.minecolonies.api.colony.interactionhandling;
+
+import com.minecolonies.api.colony.ICitizen;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.util.Tuple;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+
+/**
+ * The abstract interaction response handler to be extended by the other ones.
+ */
+public abstract class AbstractInteractionResponseHandler implements IInteractionResponseHandler
+{
+    /**
+     * The text the citizen is saying.
+     */
+    private ITextComponent inquiry;
+
+    /**
+     * The map of response options of the player, to new inquires of the interacting entity.
+     */
+    private Map<ITextComponent, ITextComponent> responses = new HashMap<>();
```

Comment:
For both things, I use ITextComponent because there is a vital difference between a request for two different pieces of armour, there is a difference between a miner complaining about a missing tool for two different locations and there is a difference between the farmer having no seed set in different fields at different positions. So we need something wrap a "key" and "a list of values" and use this as a key (uniqueness) then we have ITextComponent which does all of this already. We can use the TranslationTextComponent already here, but that wouldn't change much.